src/java/com/puppetlabs/ssl_utils/ExtensionsUtils.java

```diff
@@ -759,42 +762,64 @@ private static AuthorityKeyIdentifier mapToAuthorityKeyIdentifier(
+        if (cert == null) {
+            if (pubKey == null && serialNumber == null) {
                 throw new IllegalArgumentException(
-                        "'serial_number' not provided for auth key identifier" +
-                        "but was expected since 'issuer' was provided");
+                        "Neither 'public_key', 'serial_number', or 'cert' provided for " +
+                        "auth key identifier.  At least one of these must be " +
+                        "provided.");
             }
 
-            GeneralNames issuerAsGeneralNames =
-                    new GeneralNames(new GeneralName(new X500Name(issuer)));
-            if (authorityKeyId != null) {
-                authorityKeyId = new AuthorityKeyIdentifier(
-                        authorityKeyId.getKeyIdentifier(),
-                        issuerAsGeneralNames,
-                        serialNumber);
+            String issuer = (String) authKeyIdMap.get ("issuer_dn");
+            if (issuer == null) {
+                if (serialNumber != null) {
+                    throw new IllegalArgumentException(
+                            "'issuer' not provided for auth key identifier " +
+                            "but was expected since 'serial_number' was provided");
+                }
             }
             else {
-                authorityKeyId = new AuthorityKeyIdentifier(
-                        issuerAsGeneralNames,
-                        serialNumber);
+                if (serialNumber == null) {
+                    throw new IllegalArgumentException(
+                            "'serial_number' not provided for auth key identifier" +
+                            "but was expected since 'issuer' was provided");
+                }
+
+                GeneralNames issuerAsGeneralNames =
+                        new GeneralNames(new GeneralName(new X500Name(issuer)));
+                if (authorityKeyId != null) {
+                    authorityKeyId = new AuthorityKeyIdentifier(
+                            authorityKeyId.getKeyIdentifier(),
+                            issuerAsGeneralNames,
+                            serialNumber);
+                }
+                else {
+                    authorityKeyId = new AuthorityKeyIdentifier(
+                            issuerAsGeneralNames,
+                            serialNumber);
+                }
             }
         }
+        else {
+            DigestCalculator digCalc =
+                new JcaDigestCalculatorProviderBuilder().build().get(
+                               new AlgorithmIdentifier(OIWObjectIdentifiers.idSHA1));
+
+            X509ExtensionUtils utils = new X509ExtensionUtils(digCalc);
+            authorityKeyId = utils.createAuthorityKeyIdentifier(new X509CertificateHolder(cert.getEncoded()));
+
+            // byte[] extBytes = cert.getExtensionValue(Extension.subjectKeyIdentifier.toString());
+            // authorityKeyId = new AuthorityKeyIdentifier(extBytes);
```

Comment:
I really expected this to succeed; the key produced in the cert signed is not equivalent to the `subjectKeyIdentifier` in the ca cert, and I didn't quite know how to attempt to troubleshoot the byte String created by `getExtensionValue`.